MachineCodingRound/main.cpp

```diff
@@ -0,0 +1,136 @@
+#include <iostream>
+#include<bits/stdc++.h>
+
+using namespace std;
+
+
+class snake
+{
+public:
+    int head;
+    int tail;
+
+    snake(int headOfSnakes, int tailOfSnakes)
+    {
+        head = headOfSnakes;
+        tail = tailOfSnakes;
+    }
+
+};
+
+class ladder
+{
+public:
+    int start_positions;
+    int end_positions;
+    ladder(int start,int end)
+    {
+        start_positions = start;
+        end_positions = end;
+    }
+};
+
+class player
+{
+public:
+    string playerName;
+    int curPositions;
+    player(string name)
+    {
+        playerName = name;
+        curPositions =0;
+    }
+};
+
+class Dice {
+    public:
+        Dice(){
+             srand(time(NULL));
+        }
+    int roolDice(){
+
+        return rand()%6+1;
+    }
+};
+class snakeAndLadderGame
+{   public:
+    int numberOfSnakes;
+    int numberOfladders;
+    int numberOfPlayers;
+    vector<snake> snakes;
+    vector<ladder> ladders;
+    vector<player> players;
+    bool isGameRunning ;
+    Dice dice ;
+    void input()
```

Comment:
should not be a part of this class. Snakes, Ladders and Players should be passed either in constructor or some initializer or some setter. Do not directly take input in this class